cmd/groundstation/database/interface.go

```diff
@@ -0,0 +1,13 @@
+package database
+
+import "github.com/gpuctl/gpuctl/internal/status"
+
+// define set of operations on the database that any provider will implement
+type Database interface {
+	// add the info from a heartbeat packet to the data store
+	// that machine may have not been seen before
+	Heartbeat(packet status.GPUStatusPacket) error
```

Comment:
Wat????!!???

The heartbeat just contains "This node is up", and no GPU data. It's not a concept at the DB layer.

Also, the GPU data as stored in the Database shouldn't be related to the GPU data as tranmitted over the network.

Reply:
about the first point, i agree

about the second point, gpu data in the database is a function of the gpu data sent by our satellites, and actually we may just have a table which is verbatim the gpu stats sent to us. once schema is established we'll have more info on this but our database is likely going to be tables where either:

table.extend(f(gpu stat received))

or 

table.extend(reducer([list of gpu stats received]))

Reply:
I agree with the first point in the long term, though I think it could be useful whilst we get the system up and running to have one endpoint that pushes gpu status info into the database, then eventually split that out into heartbeats that just update us that the server is still alive and other channels for more secure data later.

As for the second point, I had reused that struct because it's the both the form the data is generated/reaches us from the satellites (https://github.com/gpuctl/gpuctl/blob/2404956106d3388dfbd1ddccacbeac00c83863fd/internal/status/gpustatus.go#L3) and the form of data that the frontend expects (https://github.com/gpuctl/gpuctl/blob/1f4d700707db9a2ad8026a19261a05a4eda1e0a8/frontend/src/App.tsx#L7), so it seemed sensible that we would want to store it in that form too